src/Widgets/GoogleMapsWidget/GoogleMapsWidgetComponent.js

```diff
@@ -56,45 +56,45 @@ function StaticMap({ address, apiKey, zoom, children }) {
   const ref = React.useRef(null);
 
   React.useEffect(() => {
+    function handleResize() {
+      const currentRef = ref.current;
+
+      if (currentRef) {
+        const elementWidth = currentRef.offsetWidth;
+        const elementHeight = currentRef.offsetHeight;
+
+        if (
+          mapBoundary.elementWidth !== elementWidth ||
+          mapBoundary.elementHeight !== elementHeight
+        ) {
+          let width = elementWidth;
+          let height = elementHeight;
+
+          if (width > maxWidth) {
+            width = maxWidth;
+
+            const factor = elementHeight / elementWidth;
+            height = Math.round(maxWidth * factor);
+          }
+
+          setMapBoundary({
+            elementHeight,
+            elementWidth,
+            height,
+            width,
+          });
+        }
+      }
+    }
+
     handleResize();
 
     window.addEventListener("resize", handleResize);
 
     return () => {
       window.removeEventListener("resize", handleResize);
     };
-  }, []);
-
-  function handleResize() {
-    const currentRef = ref.current;
-
-    if (currentRef) {
-      const elementWidth = currentRef.offsetWidth;
-      const elementHeight = currentRef.offsetHeight;
-
-      if (
-        mapBoundary.elementWidth !== elementWidth ||
-        mapBoundary.elementHeight !== elementHeight
-      ) {
-        let width = elementWidth;
-        let height = elementHeight;
-
-        if (width > maxWidth) {
-          width = maxWidth;
-
-          const factor = elementHeight / elementWidth;
-          height = Math.round(maxWidth * factor);
-        }
-
-        setMapBoundary({
-          elementHeight,
-          elementWidth,
-          height,
-          width,
-        });
-      }
-    }
-  }
+  }, [mapBoundary]);
```

Comment:
If we are rerunning the effect because the size changes, we also have to take care of removing the previous event listener.

* [ ] Q: Why is this change needed? The commit message doesn't give an explanation.

Reply:
`[mapBoundary]` is used to allow `handleResize` operates on a fresh state. With `[]`,  `handleResize` operates on a stale state with initial values. And we want to reduce the amount of invocation of `setMapBoundary`, with this condition ```
        if (
          mapBoundary.elementWidth !== elementWidth ||
          mapBoundary.elementHeight !== elementHeight
        ) ```

The previous event listener is removed in useEffect's return function. I've checked if there is any leak and I didn't noticed any problems:
```
> window.getEventListeners(window)
{message: Array(3), focus: Array(1), resize: Array(1), beforeunload: Array(1)}
```